### PR TITLE
:warning: Use multi-AZ as default example configuration 

### PR DESCRIPTION
**Pull request by Sn0rt**
In most scenarios with availability requirements, customers will create clusters that fully support full-region fault tolerance.

TODO: Multiple private subnets share one natgw.

#967 

Signed-off-by: guohaowang <wangguohao.2009@gmail.com>

### CODE REVIEW

File: examples/cluster/cluster.yaml [hunk 1]
```diff
@@ -6,7 +6,11 @@ metadata:
 spec:
   clusterNetwork:
     pods:
-      cidrBlocks: ["192.168.0.0/16"]
+      cidrBlocks:
+        - "10.0.0.0/8"
+    services:
+      cidrBlocks:
+        - "192.168.0.0/16"
```
Comment (vincepri) [inline examples/cluster/cluster.yaml]: Are these changes necessary and related to this PR?

File: examples/cluster/cluster.yaml [hunk 2]
```diff
@@ -19,3 +23,23 @@ metadata:
 spec:
   region: ${AWS_REGION}
   sshKeyName: ${SSH_KEY_NAME}
+  networkSpec:
+    subnets:
+    - availabilityZone: ${AWS_REGION}a
+      cidrBlock: "10.0.100.0/24"
```
Comment (vincepri) [inline examples/cluster/cluster.yaml]: Given that we use `10.0.0.0/16` by default, can we use the first subnets in the range http://www.davidc.net/sites/default/subnets/subnets.html?network=10.0.0.0&mask=16&division=23.ff4600 ?

**Comment by vincepri** [comment]
/ok-to-test

Retitled to show that this is actually a breaking change in the manifest generation

File: examples/cluster/cluster.yaml (hunk 2 above)
Comment (detiber) [reply examples/cluster/cluster.yaml]: We really shouldn't overlap this with the pod network.

File: examples/controlplane/controlplane.yaml [hunk 3]
```diff
@@ -25,6 +25,7 @@ spec:
   instanceType: ${CONTROL_PLANE_MACHINE_TYPE}
   iamInstanceProfile: "control-plane.cluster-api-provider-aws.sigs.k8s.io"
   sshKeyName: "${SSH_KEY_NAME}"
+  availabilityZone: ${AWS_REGION}a
```
Comment (detiber) [inline examples/controlplane/controlplane.yaml]: Not every region has a `a`, `b`, and `c` AZ (nor do they necessarily align between different accounts). Hardcoding these here may lead to non-functioning manifests for some users.

File: examples/cluster/cluster.yaml (hunk 1 above)
Comment (detiber) [reply examples/cluster/cluster.yaml]: This will break the quick start, which is based on calico today.

**Comment by ncdc** [comment]
Is this related to #930?

File: examples/cluster/cluster.yaml (hunk 1 above)
Comment (Sn0rt) [reply examples/cluster/cluster.yaml]: > This will break the quick start, which is based on calico today.

oho, this is my mistake, this setting just for amazon-vpc-cni. if user want to use calico , the CIDR of pods are different between vpc's.

File: examples/cluster/cluster.yaml (hunk 2 above)
Comment (Sn0rt) [reply examples/cluster/cluster.yaml]: > See above, I think the changes to the cluster configuration aren't necessary

yep, in fact, I hope that the generate script can generate the ha cluster yaml with amazon-vpc-cni or with  calico

File: examples/controlplane/controlplane.yaml (hunk 3 above)
Comment (Sn0rt) [reply examples/controlplane/controlplane.yaml]: > Not every region has a `a`, `b`, and `c` AZ (nor do they necessarily align between different accounts). Hardcoding these here may lead to non-functioning manifests for some users.

right, I will improve this script if this region has the a, b, c AZ , the script will generate the cluster yaml as before, if not, the script will set three masters with one az. what did you think?

Comment (Sn0rt) [reply examples/controlplane/controlplane.yaml]: the az info as follows

eu-north-1: eu-north-1a eu-north-1b eu-north-1c
ap-south-1: ap-south-1a ap-south-1b ap-south-1c
eu-west-3: eu-west-3a eu-west-3b eu-west-3c
eu-west-2: eu-west-2a eu-west-2b eu-west-2c
eu-west-1: eu-west-1a eu-west-1b eu-west-1c
ap-northeast-2: ap-northeast-2a ap-northeast-2b ap-northeast-2c
ap-northeast-1: ap-northeast-1a ap-northeast-1c ap-northeast-1d
sa-east-1: sa-east-1a sa-east-1b sa-east-1c
ca-central-1: ca-central-1a ca-central-1b
ap-east-1: ap-east-1a ap-east-1b ap-east-1c
ap-southeast-1: ap-southeast-1a ap-southeast-1b ap-southeast-1c
ap-southeast-2: ap-southeast-2a ap-southeast-2b ap-southeast-2c
eu-central-1: eu-central-1a eu-central-1b eu-central-1c
us-east-1: us-east-1a us-east-1b us-east-1c us-east-1d us-east-1e us-east-1f
us-east-2: us-east-2a us-east-2b us-east-2c
us-west-1: us-west-1b us-west-1c
us-west-2: us-west-2a us-west-2b us-west-2c us-west-2d

**Comment by ncdc** [comment]
Thanks for putting this together. We want to reevaluate how we support different flavors of cluster configurations after we have clusterctl v2 in place. For now, we're going to close this.

/close
